apps/full-site-editing/full-site-editing-plugin/full-site-editing/plugins/close-button-override/index.js

```diff
@@ -4,28 +4,42 @@
  * External dependencies
  */
 import domReady from '@wordpress/dom-ready';
+import { __ } from '@wordpress/i18n';
+import './style.scss';
 
 domReady( () => {
 	const { closeButtonLabel, closeButtonUrl, editorPostType } = fullSiteEditing;
 
 	const editPostHeaderInception = setInterval( () => {
-		const closeButton = document.querySelector( '.edit-post-fullscreen-mode-close__toolbar a' );
+		// cycle through interval until header toolbar is found
+		const toolbar = document.querySelector( '.edit-post-header__toolbar' );
 
-		if ( ! closeButton ) {
+		if ( ! toolbar ) {
 			return;
 		}
 		clearInterval( editPostHeaderInception );
 
+		// add components toolbar with override class name (original will be hidden in ./style.scss)
+		const componentsToolbar = document.createElement( 'div' );
+		componentsToolbar.className =
+			'components-toolbar edit-post-fullscreen-mode-close__toolbar edit-post-fullscreen-mode-close__toolbar__override';
+		toolbar.prepend( componentsToolbar );
+
+		// create custom close button and append to components toolbar
+		const newCloseButton = document.createElement( 'a' );
 		// When closing Template CPT (e.g. header) to navigate back to parent page.
 		if ( 'wp_template_part' === editorPostType && closeButtonUrl ) {
-			const newCloseButton = document.createElement( 'a' );
 			newCloseButton.href = closeButtonUrl;
 			newCloseButton.innerHTML = closeButtonLabel;
-			newCloseButton.className = 'components-button components-icon-button is-button is-default';
 			newCloseButton.setAttribute( 'aria-label', closeButtonLabel );
-
-			const parentContainer = document.querySelector( '.edit-post-fullscreen-mode-close__toolbar' );
-			parentContainer.replaceChild( newCloseButton, closeButton );
+		} else if ( 'page' === editorPostType ) {
+			newCloseButton.href = 'edit.php?post_type=page';
+			const newLabel = __( 'Back to Page List' );
```

Comment:
Similar to above, maybe `closeButtonLabel` would work in this case. To clarify, both `closeButtonLabel` and `closeButtonUrl` are populated to the `fullSiteEditing` global object in PHP when the scripts are enqueued here: https://github.com/Automattic/wp-calypso/blob/56b34e777a027be32e8e3051c274fb53c66c4458/apps/full-site-editing/full-site-editing-plugin/full-site-editing/class-full-site-editing.php#L167-L176)

Reply:
Also, thanks for remembering I18n here!

Reply:
Unfortunately the `closeButtonLabel` comes out as `null` in the context of the page editor as well.

Reply:
hm I would have expected it to be there. Weird. Not a big deal though. At some point, we should really create a robust navigation solution for this instead of hacking at the DOM (but that's not our goal for today :) )